### PR TITLE
Don't use HttpServerTest.controller in java tests

### PR DESCRIPTION
**Pull request by laurit**
`HttpServerTest` is a groovy class, we should use it from the java tests

### CODE REVIEW

File: instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v3_1/filter/ServletFilterConfig.java [hunk 1]
```diff
@@ -29,53 +29,43 @@ Filter servletFilter() {
...
-            () -> {
-              resp.setContentType("text/plain");
-              switch (endpoint.name()) {
-                case "SUCCESS":
-                  resp.setStatus(endpoint.getStatus());
-                  resp.getWriter().print(endpoint.getBody());
-                  break;
-                case "QUERY_PARAM":
+        if (endpoint != null && endpoint != ServerEndpoint.NOT_FOUND) {
+          AbstractHttpServerTest.controller(
```
Comment (jaydeluca) [inline instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v3_1/filter/ServletFilterConfig.java:41]: static import here?

File: instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/filter/ServletFilterConfig.java [hunk 2]
```diff
@@ -29,53 +29,43 @@ Filter servletFilter() {
...
-            () -> {
-              resp.setContentType("text/plain");
-              switch (endpoint.name()) {
-                case "SUCCESS":
-                  resp.setStatus(endpoint.getStatus());
-                  resp.getWriter().print(endpoint.getBody());
-                  break;
-                case "QUERY_PARAM":
+        if (endpoint != null && endpoint != ServerEndpoint.NOT_FOUND) {
+          AbstractHttpServerTest.controller(
```
Comment (jaydeluca) [inline instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/filter/ServletFilterConfig.java:41]:
```suggestion
         controller(
```

File: instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/GreetingAction.java [hunk 3]
```diff
@@ -17,40 +19,37 @@ public class GreetingAction extends ActionSupport {
...
         () -> {
-          throw new Exception(ServerEndpoint.EXCEPTION.getBody());
+          throw new IllegalStateException(ServerEndpoint.EXCEPTION.getBody());
         });
     throw new AssertionError(); // should not reach here
   }
 
   public String path_param() {
-    HttpServerTest.controller(
+    controller(
```
Comment (jaydeluca) [inline instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/GreetingAction.java:52]: there are two more occurrences of `HttpServerTest.controller` in this file, should we also replace those?

File: instrumentation/tomcat/tomcat-10.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v10_0/TestServlet.java [hunk 4]
```diff
@@ -21,11 +20,11 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
 
     ServerEndpoint serverEndpoint = ServerEndpoint.forPath(path);
     if (serverEndpoint != null) {
-      HttpServerTest.controller(
+      AbstractHttpServerTest.controller(
```
Comment (jaydeluca) [inline instrumentation/tomcat/tomcat-10.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v10_0/TestServlet.java:23]:
```suggestion
     controller(
```

File: instrumentation/tomcat/tomcat-7.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v7_0/AsyncServlet.java [hunk 5]
```diff
@@ -37,7 +36,7 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) {
     context.start(
         () -> {
           try {
-            HttpServerTest.controller(
+            AbstractHttpServerTest.controller(
```
Comment (jaydeluca) [inline instrumentation/tomcat/tomcat-7.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v7_0/AsyncServlet.java:39]:
```suggestion
           controller(
```

File: instrumentation/tomcat/tomcat-7.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v7_0/TestServlet.java [hunk 6]
```diff
@@ -21,11 +20,11 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
 
     ServerEndpoint serverEndpoint = ServerEndpoint.forPath(path);
     if (serverEndpoint != null) {
-      HttpServerTest.controller(
+      AbstractHttpServerTest.controller(
```
Comment (jaydeluca) [inline instrumentation/tomcat/tomcat-7.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/v7_0/TestServlet.java:23]:
```suggestion
     controller(
```

**Review by steverao — COMMENTED** [review]
There are some places that you seem forgot to modify them together, such as https://github.com/laurit/opentelemetry-java-instrumentation/blob/978ae8e900b85429257c491acfd7af460bc4276f/instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/GreetingAction.java#L61

File: instrumentation/servlet/servlet-2.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v2_2/TestServlet2.java [hunk 7]
```diff
@@ -24,10 +25,10 @@ private TestServlet2() {}
 
   public static class Sync extends HttpServlet {
     @Override
-    protected void service(HttpServletRequest req, HttpServletResponse resp) {
+    protected void service(HttpServletRequest req, HttpServletResponse resp) throws IOException {
       req.getRequestDispatcher(null);
       ServerEndpoint endpoint = ServerEndpoint.forPath(req.getServletPath());
-      HttpServerTest.controller(
+      AbstractHttpServerTest.controller(
```
Comment (steverao) [inline instrumentation/servlet/servlet-2.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v2_2/TestServlet2.java:31]:
```suggestion
      controller(
```

File: instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/GreetingAction.java (hunk 3 above)
Comment (laurit) [reply instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/struts2/GreetingAction.java:51]: thanks

File: instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v3_1/filter/ServletFilterConfig.java (hunk 1 above)
Comment (laurit) [reply instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v3_1/filter/ServletFilterConfig.java:41]: not sure the static import is worth it if there is only 1 usage

**Pull request merged**